packages/plugin-ext/src/main/browser/metadata-selection.ts

```diff
@@ -0,0 +1,51 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+export interface MetadataSelection {
+    // tslint:disable-next-line:no-any
+    readonly metadata: any
+}
+
+export namespace MetadataSelection {
+
+    export function is(arg: Object | undefined): arg is MetadataSelection {
+        // tslint:disable-next-line:no-any
+        return typeof arg === 'object' && ('metadata' in arg);
+    }
+
+    // tslint:disable-next-line:no-any
+    export function getMetadata(selection: Object | undefined): any | undefined {
+        if (is(selection)) {
+            return selection.metadata;
+        }
+        if (Array.isArray(selection) && is(selection[0])) {
```

Comment:
Does VS Code allow multi selection in contributed tree views?

Reply:
Have no idea, looking for it now. But meantime, we're allowing to create tree widgets that allow use multi selection. (via tree properties)

Reply:
Just checked, that they allow, but in the context menu only the node under selection is provided, so it is good how you do it

Reply:
> we're allowing to create tree widgets that allow use multi selection. (via tree properties)

I cannot find it. For me multi-selection in the tree widget does not work in Theia, but in VS Code work. I check with `Node Dependencies` tree from here: https://github.com/akosyakov/vscode-trees

Reply:
I'm not 100% sure about it, but I made this conclusion based on [this](https://github.com/theia-ide/theia/blob/master/packages/core/src/browser/tree/tree-widget.tsx#L70) property, which pass into tree widget constructor [here](https://github.com/theia-ide/theia/blob/master/packages/core/src/browser/tree/tree-widget.tsx#L130), this just need to be checked.

> Could it be that multi-selection should be enabled by default?

I think it depends on configuration for particular use case. For example in kubernetes plugin there is a single selection option, but in file navigator a multi selection enabled. So I think, that multi selection might be enabled by default.